services/ssh/home/rsh.sh

```diff
@@ -120,6 +124,36 @@ fi
 
 # If there is a deployment for the given service searching for lagoon.sh labels
 if [[ $($OC get deployment -l "lagoon.sh/service=${SERVICE}" 2> /dev/null) ]]; then
+  # get any other deployments that may have been idled by the idler and unidle them if required
+  # this only needs to be done for kubernetes
+  # we do this first to give the services a bit of time to unidle before starting the one that was requested
+  DEPLOYMENTS=$($OC get deployments -l "idling.amazee.io/watch=true" -o name)
+  if [ ! -z "${DEPLOYMENTS}" ]; then
+    # loop over the deployments and unidle them
+    for DEP in ${DEPLOYMENTS}
+    do
+      # if the deployment is idled, unidle it :)
+      DEP_JSON=$($OC get ${DEP} -o json)
+      if [ $(echo "$DEP_JSON" | jq -r '.status.replicas // 0') == "0" ]; then
+        REPLICAS=$(echo "$DEP_JSON" | jq -r '.metadata.annotations."idling.amazee.io/unidle-replicas" // 1')
+        if [ ! -z "$REPLICAS" ]; then
+          REPLICAS=1
+        fi
+        $OC scale --replicas=${REPLICAS} ${DEP} >/dev/null 2>&1
+        # for unidling an entire environment and waiting for the number of `readyReplicas`
+        # to be 1 for each deployment, could add considerable delays for the ssh connection to establish.
+        # WAIT_TO_UNIDLE_SERVICES will default to false so that it just scales the deployments
+        # and won't wait for them to be ready, but if set to true, it will wait for `readyReplicas` to be 1
+        if [[ "$WAIT_TO_UNIDLE_SERVICES" =~ [Tt][Rr][Uu][Ee] ]]; then
+          while [[ ! $($OC get ${DEP} -o go-template --template='{{.status.readyReplicas}}') == "1" ]]
+          do
+            sleep 1
+          done
+        fi
```

Comment:
I would suggest two changes:
1. can we split the actual unidling and the waiting into two for loops? In this case we first trigger scaling of all deployments and then maybe wait for them to be available, which would probably reduce the time that we need to wait.
2. if a deployment had 2 replicas, the `--template='{{.status.readyReplicas}}') == "1"` will fail, I think we should just check that is has 1 or more replicas.

Reply:
On 1) waiting for all services in the environment to become available could add considerable delays to establishing the ssh connection though. Which is why I did it the way I did, where it unidles them but won't wait for them to become ready (unless the `WAIT_TO_UNIDLE_SERVICES` var is set to true, which defaults to false)

I'll fix the 1+ replica check now.